daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -216,3 +236,79 @@ runLedgerNavigator flags remainingArguments = do
           ]
         , ["  }"]
         ]
+
+----------------
+-- HTTP JSON API
+----------------
+
+newtype Method = Method
+  { unMethod :: BS.ByteString
+  } deriving IsString
+
+newtype Path = Path
+  { unPath :: BS.ByteString
+  } deriving IsString
+
+-- | Run a request against the HTTP JSON API.
+httpJsonRequest :: LedgerArgs -> Method -> Path -> IO (Response A.Value)
+httpJsonRequest LedgerArgs {..} method path = do
+  resp <-
+    httpJSON $
+    setRequestPort port $
+    setRequestHost (BSC.pack host) $
+    setRequestMethod (unMethod method) $
+    setRequestPath (unPath path) $
```

Comment:
The logic here only works if the JSON API is hosted at `/` and not a subpath. I believe we have this limitation in other places (e.g., DAML Script) so not a big issue but worth keeping in mind.

Reply:
Does that work with DABL?

Reply:
Not quite sure where dabl hosts the API but afaik they don’t expose the admin endpoints anyway.